provisioning/device/tests/ArgumentTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.UnitTests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class ArgumentTests
+    {
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void AssertNotNull_IfNull_Throws()
+        {
+            string argument = null;
+            Argument.AssertNotNull<string>(argument, "");
```

Comment:
Just curious why we are not using FluentAssertions?

Reply:
Curious about this too. We have both ways of doing it present in our tests, but is this something where we should be sticking with one style?

Reply:
Argument.AssertNotNull is not a unit test assertion framework. We use that for validating parameters in our public API.